keps/sig-instrumentation/3077-contextual-logging/README.md

```diff
@@ -870,7 +912,7 @@ logging.
 
 #### Beta
 
-- All of kube-scheduler (in-tree) and CSI external-provisioner (out-of-tree) converted
+- [All of kube-controller-manager](https://github.com/kubernetes/kubernetes/pull/119250) and some [parts of kube-scheduler](https://github.com/kubernetes/kubernetes/pull/115588) converted (in-tree), conversion of out-of-tree components possible, whether they use pflag ([external-provisioner](https://github.com/kubernetes-csi/external-provisioner/pull/639)] or plain Go flags ([node-driver-registrar](https://github.com/kubernetes-csi/node-driver-registrar/pull/259))
```

Comment:
We ended up focusing on kube-controller-manager instead of kube-scheduler because that work was easier to split up. Conversion of external components is technically feasible, but was delayed by lack of developer time. Contributors are willing to take on that part now and have already started ([1](https://github.com/kubernetes-csi/csi-lib-utils/pull/149), [2](https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/pull/154)).

Regarding the next points (cannot comment on it in GitHub):
> Gathered feedback from developers and surveys

There has been steady interest in this, as seen from questions on Slack and good attendance of the KubeCon maintainer track sessions about logging. We've not done a survey, and I am not sure what exactly I would ask in it. May I perhaps remove that part?

> New APIs in `k8s.io/klog/v2` no longer marked as experimental

Not done yet, will do before the klog update for 1.29. Those new APIs haven't changed for several releases now and can be promoted to stable.

Reply:
Do you have any estimates around how much work is left in kube-scheduler? Also what about kubelet and kube-apiserver and cloud-controller-manager? I'd like to see explicit information about each at beta.

Reply:
I'm tracking that in the description of https://github.com/kubernetes/enhancements/issues/3077.

Shall I copy a snapshot of that information into the KEP?

kube-scheduler and kube-controller-manager are done and only need to be touch again when we make API changes elsewhere,  for example in component-base or client-go.

kubelet has around 2000 calls which need to be updated, kube-proxy 400, kube-apiserver 660.

I've not checked cmd/cloud-controller-manager, but it seems to be small.

Reply:
That's fine, I didn't look there. No need to put that information in the KEP in that case.